permissions/rules/za.js

```diff
@@ -19,6 +19,14 @@ allow(
 
 allow(User, Actions.REOPEN_CASE, Case, user => isSupervisor(user));
 
+allow(
+  User,
+  Actions.CASE_STATUS_TRANSITION,
+  Case,
+  (user, caseObj) =>
```

Comment:
`(user, caseObj) => isSupervisor(user) || (isCaseOpen(caseObj) && isCounselorWhoCreated(user, caseObj))` is repeated many times across our rules.  Would we want to extract that out into a helper?

Reply:
Agreed. Is `isSupervisorOrCaseCreator` too long for naming this helper?

Reply:
It's not too long, but it could be misleading because the case needs to be open for the creator to work it.

Reply:
I actually wouldn't worry too much about refactoring it right now, because at some point we will move to fetch rules from a JSON, so we won't be writing these rules ourselves.